src/core/space_interaction_API.jl

```diff
@@ -0,0 +1,186 @@
+#=
+This file establishes the agent-space interaction API.
+All space types should implement this API.
+Some functions DO NOT need to be implemented for every space, they are space agnostic.
+These functions have complete source code here, while the functions that DO need to
+be implemented for every space have only documentation strings here and an
+error message.
+
+In short: IMPLEMENT ALL FUNCTIONS IN SECTION "IMPLEMENT", WITH SAME ARGUMENTS!
+=#
+export move_agent!,
+    add_agent!,
+    add_agent_pos!,
+    kill_agent!,
+    genocide!
+
+notimplemented(model) = error("Not implemented for space type $(nameof(typeof(model.space)))")
+
+#######################################################################################
+# %% IMPLEMENT
+#######################################################################################
+"""
+    random_position(model) → pos
+Return a random position in the model's space (always with appropriate Type).
+"""
+random_position(model) = notimplemented(model)
+
+"""
+    move_agent!(agent [, pos], model::ABM) → agent
+
+Move agent to the given position, or to a random one if a position is not given.
+`pos` must be the appropriate position type depending on the space type.
+"""
+move_agent!(agent::A, pos::ValidPos, model::ABM) where {A<:AbstractAgent} = notimplemented(model)
+
+"""
+    add_agent_to_space!(agent, model)
+Add the agent to the underlying space structure at the agent's own position.
+This function is called after the agent is already inserted into the model dictionary
+and changing max id has been taken care of. This function is NOT part of the public API.
```

Comment:
Suggestion
and changing max id has been taken care of -> and the `maxid` has been updated